lib/dl_core_testing/dl_core_testing/testcases/adapter.py

```diff
@@ -0,0 +1,87 @@
+import abc
+from typing import (
+    ClassVar,
+    Generic,
+    Type,
+    TypeVar,
+)
+
+import pytest
+
+from dl_api_commons.base_models import RequestContextInfo
+from dl_core import exc
+from dl_core.connection_executors import AsyncConnExecutorBase
+from dl_core.connection_executors.adapters.async_adapters_base import AsyncDirectDBAdapter
+from dl_core.connection_executors.models.connection_target_dto_base import BaseSQLConnTargetDTO
+from dl_core.connection_executors.models.db_adapter_data import DBAdapterQuery
+from dl_core.connection_executors.models.scoped_rci import DBAdapterScopedRCI
+from dl_core_testing.testcases.connection_executor import BaseConnectionExecutorTestClass
+
+
+_TARGET_DTO_TV = TypeVar("_TARGET_DTO_TV", bound=BaseSQLConnTargetDTO)
+
+
+class BaseAsyncAdapterTestClass(BaseConnectionExecutorTestClass, Generic[_TARGET_DTO_TV], metaclass=abc.ABCMeta):
+    """
+    Most of the testable adapter behaviour is covered by conn executor tests
+    Here are the tests, that are just easier to implement at the adapter level
+    """
+
+    ASYNC_ADAPTER_CLS: ClassVar[Type[AsyncDirectDBAdapter]]  # TODO add tests for other adapters
+
+    @pytest.fixture
+    async def target_conn_dto(self, async_connection_executor: AsyncConnExecutorBase) -> _TARGET_DTO_TV:
+        return next(iter(await async_connection_executor._make_target_conn_dto_pool()))  # noqa
+
+    def _make_dba(self, target_dto: _TARGET_DTO_TV, rci: RequestContextInfo) -> AsyncDirectDBAdapter:
+        return self.ASYNC_ADAPTER_CLS.create(
+            req_ctx_info=DBAdapterScopedRCI.from_full_rci(rci),
+            target_dto=target_dto,
+            default_chunk_size=10,
+        )
+
+    test_debug_query_with_default_value: ClassVar[bool] = False
```

Comment:
please use `RegulatedTestCase` instead of classvars
if there is a chance this classvar would be used for more base tests, create a feature for this, like [array_support](https://github.com/datalens-tech/datalens-backend/blob/35a63b2a3126c0e9634b781107b0282dca2032ac/lib/dl_api_lib_testing/dl_api_lib_testing/connector/data_api_suites.py#L35-L35)

Reply:
I can't see how I can utilize `RegulatedTestCase`/`for_feature` to skip a single case of a parametrized test. Can you show me?

Reply:
hmm, i missed the parametrization part. on top of my head there are two ways to deal with it:
1. robust: split the test in two and move the common part in a separate function, somewhat like [this](http://github.com/datalens-tech/datalens-backend/blob/35a63b2a3126c0e9634b781107b0282dca2032ac/lib/dl_api_lib_testing/dl_api_lib_testing/connector/data_api_suites.py#L119-L143). this way you can ignore just the `pass_db_query_to_user is None` case
2. flexible: implement this functionality in `RegulatedTestCase`. shouldn't be too difficult, instead of the [skip wrapper](http://github.com/datalens-tech/datalens-backend/blob/35a63b2a3126c0e9634b781107b0282dca2032ac/lib/dl_testing/dl_testing/regulated_test.py#L60-L68) create a skipif version.

i think the first option is okay (and probably preferable) for now